src/classes/routine-service-proxy/routine-service-proxy.cpp

```diff
@@ -0,0 +1,76 @@
+#include "routine-service-proxy.hpp"
+
+RoutineServiceProxy::RoutineServiceProxy(RoutineService& realService, std::map<std::string, ServiceState::state_t> depsMap)
+    : ServiceProxy(realService, ROUTINE_SERVICE, depsMap)
+{
+    changeState(ServiceState::MISSING_DEPENDENCIES);
+}
+
+void RoutineServiceProxy::MissingDependencies::allFine()
+{
+    // Expose/Hide the endpoint on the DBUS and THEN..
+    m_upperProxy.weave();
+}
+
+void RoutineServiceProxy::Running::somethingIsMissing()
+{
+    // Unexpose/Hide the endpoint on the DBUS and THEN..
+    m_upperProxy.cut();
+}
+
+void RoutineServiceProxy::serviceCycle()
+{
+    changeState(ServiceState::RUNNING);
+    m_realService.setup();
+
+    while (checkState() != ServiceState::STOPPED) {
+
+        m_realService.routine();
+    }
+
+    m_realService.destroy();
+
+    changeState(ServiceState::FINISHED);
+}
+
+void RoutineServiceProxy::weave()
+{
+    const std::lock_guard<std::mutex> lock(m_updateMtx);
```

Comment:
```suggestion
    const std::lock_guard<std::mutex> lock{m_updateMtx};
```

Em geral, isso melhora a legibilidade do código.

Reply:
Legibilidade melhorada!!